sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala

```diff
@@ -253,7 +253,8 @@ case class StateStoreRestoreExec(
       stateManager.getStateValueSchema,
       indexOrdinal = None,
       sqlContext.sessionState,
-      Some(sqlContext.streams.stateStoreCoordinator)) { case (store, iter) =>
+      Some(sqlContext.streams.stateStoreCoordinator),
+      readOnly = true) { case (store, iter) =>
```

Comment:
Is this only place we use read-only state store?

Reply:
Yes. All other cases do read and write in same place. Streaming aggregation handles aggregation as `do aggregation in each partition` -> `shuffle` -> `enrich accumulated values via loading from state store` -> do aggregation -> `store new accumulated values to state store` hence two separated places for read and write.